#!/usr/bin/env python# -*- coding: utf-8 -*-# @Time    : 2018/5/26 下午10:50# @Author  : ziyi# @File    : connec_mysql.py# @Software: PyCharmimport pymysql# 创建数据库def cre_db(host, user, pw, name):    try:        # 数据库连接        db = pymysql.connect(host, user, pw, use_unicode=True, charset='utf8')        # 创建游标，通过连接与数据通信        cursor = db.cursor()        # 执行sql语句        cursor.execute('show databases')        rows = cursor.fetchall()        for row in rows:            tmp = "{}".format(row)            print(tmp)        if name in rows:            print('数据库已经存在')            # cursor.execute('drop database if exists ' + name)        else:            print('开始创建数据库')            cursor.execute('create database if not exists ' + name)            # 提交到数据库执行            db.commit()    except pymysql.Error as e:        print("Mysql Error {}: {}".format(e.args[0], e.args[1]))    finally:        # 关闭数据库连接        db.close()def create_table(db_host, db_user, db_password, db_database):    try:        db = pymysql.connect(db_host, db_user, db_password, db_database, use_unicode=True, charset='utf8mb4')        cursor = db.cursor()        cursor.execute('DROP TABLE IF EXISTS user_infor')        create_sql = """CREATE TABLE user_infor (roomname CHAR(255), id CHAR(255) NOT NULL, nickname CHAR(255),         province CHAR(255),city CHAR(255), SEX CHAR (4),PRIMARY KEY (id))"""        cursor.execute(create_sql)        # 关闭数据库连接        db.close()    except Exception as e:        print(e)def create_table1(db_host, db_user, db_password, db_database,table_name):    try:        db = pymysql.connect(db_host, db_user, db_password, db_database, use_unicode=True, charset='utf8')        cursor = db.cursor()        # cursor.execute('DROP TABLE IF EXISTS {}'.format(table_name))        create_sql = """CREATE TABLE {} (roomname CHAR(255), id CHAR(255) NOT NULL, nickname CHAR(255),         province CHAR(255),city CHAR(255), SEX CHAR (4)) """.format(table_name)        cursor.execute(create_sql)        print('新建数据表成功')        # 关闭数据库连接        db.close()    except Exception as e:        print(e)def insert_data1(db_host, db_user, db_password, db_database,table_name):    db = pymysql.connect(db_host, db_user, db_password, db_database, use_unicode=True, charset='utf8mb4')    cursor = db.cursor()    insert_sql = """INSERT INTO {}(roomname,            id, nickname, province, city,SEX) VALUES ('nus 秋季群', '@a31e92631c6586917965e3e7cf5c9d475a4e3311',             '爱吃番茄的徐雪凡🍅', '江苏','南京', '0')""".format(table_name)    try:        # 执行sql语句        cursor.execute(insert_sql)        print('插入数据')        # 提交到数据库执行        db.commit()    except Exception as e:        # 如果发生错误则回滚        print('没有插入数据,可能有重复',e)        db.rollback()    db.close()    # sql ="""ALTER TABLE user_infor CONVERT TO CHARACTER SET utf8mb4"""    # cursor.execute(sql)    # print('修改编码')    # # 提交到数据库执行    # db.commit()if __name__ == '__main__':    db_host = 'localhost'    db_user = 'root'    db_password = 'woaini36588'    db_database = 'userdb'    table_name = 'all_user'    # cre_db(db_host, db_user, db_password, db_database)    # 新建表    create_table1(db_host, db_user, db_password, db_database, table_name)    insert_data1(db_host, db_user, db_password, db_database, table_name)